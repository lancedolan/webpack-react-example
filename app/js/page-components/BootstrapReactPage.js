import React from 'react'
import NavBar from "../components/NavBar";
import ClickCountButton from "../components/ClickCountButton";

function BootstrapReactPage(props) {

    const container =
        <div className="container" style={{marginTop: '4em'}} key="2">
            <div className="row">
                <div className="col">
                    <div className="d-grid gap-2 col-6 mx-auto">
                        <ClickCountButton />
                    </div>
                </div>
            </div>
        </div>

    return [<NavBar key="1" />,container]
}

export default BootstrapReactPage
