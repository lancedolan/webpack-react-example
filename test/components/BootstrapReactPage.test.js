import React from "react";
import render from "react-test-renderer";
import BootstrapReactPage from "../../app/js/page-components/BootstrapReactPage";

describe('BootstrapReactPage',()=>{
    const comp = render.create(<BootstrapReactPage />)
    const compJson = comp.toJSON()

    it('should be an array of non-empty components', ()=>{
        for(let comp of compJson) {
            expect(comp.children.length).toBeGreaterThan(0)
        }
    })

    it('should match snapshot', ()=>{
        expect(compJson).toMatchSnapshot()
    })

})
