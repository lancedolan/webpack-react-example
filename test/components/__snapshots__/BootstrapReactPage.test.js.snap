// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BootstrapReactPage should match snapshot 1`] = `
Array [
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <div
      className="container-fluid"
    >
      <a
        className="navbar-brand"
        href="#"
      >
        <img
          alt=""
          className="d-inline-block align-text-top"
          height="24"
          src={Object {}}
          width="24"
        />
         My App
      </a>
      <button
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-bs-target="#navbarSupportedContent"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav me-auto mb-2 mb-lg-0"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current="page"
              className="nav-link active"
              href="./index.html"
            >
              React Page
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="nav-link"
              href="./no-react.html"
            >
              Vanilla JS Page
            </a>
          </li>
          <li
            className="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              className="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              Dropdown
            </a>
            <ul
              aria-labelledby="navbarDropdown"
              className="dropdown-menu"
            >
              <li>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Dummy Link 1
                </a>
              </li>
              <li>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Dummy Link 2
                </a>
              </li>
              <li>
                <hr
                  className="dropdown-divider"
                />
              </li>
              <li>
                <a
                  className="dropdown-item"
                  href="#"
                >
                  Truly Dumb Link
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </nav>,
  <div
    className="container"
    style={
      Object {
        "marginTop": "4em",
      }
    }
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="d-grid gap-2 col-6 mx-auto"
        >
          <button
            className="btn btn-primary"
            onClick={[Function]}
            type="button"
          >
            <span>
              I've never been clicked!
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
